s3transfer/compat.py

```diff
@@ -50,3 +50,27 @@ def accepts_kwargs(func):
         return inspect.getargspec(func)[2]
 
     SOCKET_ERROR = socket.error
+
+
+def seekable(fileobj):
+    """Backwards compat function to determine if a fileobj is seekable
+
+    :param fileobj: The file-like object to determine if seekable
+
+    :returns: True, if seekable. False, otherwise.
+    """
+    try:
+        # If the fileobj has a seekable attr, try calling the seekable()
+        # method on it.
+        if hasattr(fileobj, 'seekable'):
+            return fileobj.seekable()
```

Comment:
Can seekable actually raise an IOError/OSError?  Does it need to be under the try?